### DISCUSSION TITLE
User is not sudo in Linux Workstation

### DISCUSSION THREAD

**Discussion by thiagodalsoto** [rustdesk/rustdesk#9374] [Q&A]
I am installing rustdesk on a Linux workstation where the user does not belong to the sudo group in /etc/groups.

This is necessary to restrict the user's powers locally, but I noticed that this causes some errors, for example, when logging into the system, rustdesk does not open in the tray, unless I manually open the application. Of course, this is only visual because I can connect to the workstation, but if I connect remotely and need to open rustdesk to change some configuration, I cannot. I need to ask the user to open rustdesk before I connect, so that I can change the configurations.

I have tried approaches such as;
in /etc/sudoers
ALL ALL=(ALL) NOPASSWD: /usr/lib/rustdesk/rustdesk --tray

but to no avail. Is there any way to get around this so that rustdesk runs perfectly without the user being part of sudo? 
Thanks for any help.

**rustdesk[4836]: Sorry, user l200comercial01 is not allowed to execute '/usr/lib/rustdesk/rustdesk --tray'**
**rustdesk[2874]: Sorry, user gdm is not allowed to execute '/usr/lib/rustdesk/rustdesk --tray'**

```
suporte@l200-com01:~$ sudo systemctl status rustdesk
● rustdesk.service - RustDesk
     Loaded: loaded (/usr/lib/systemd/system/rustdesk.service; enabled; preset: enabled)
     Active: active (running) since Mon 2024-09-16 13:29:42 -04; 1min 47s ago
   Main PID: 1200 (rustdesk)
      Tasks: 32 (limit: 4350)
     Memory: 57.1M (peak: 64.9M)
        CPU: 13.376s
     CGroup: /system.slice/rustdesk.service
             ├─1200 /usr/bin/rustdesk --service
             ├─4830 sudo -E XDG_RUNTIME_DIR=/run/user/632801162 -u l200comercial01 /usr/lib/rustdesk/rustdesk --server
             └─4831 /usr/lib/rustdesk/rustdesk --server

set 16 13:29:51 l200-com01.vr.local sudo[2874]:      gdm : command not allowed ; PWD=/ ; USER=gdm ; ENV=XDG_RUNTIME_DIR=/run/user/120 ; COMMAND=/usr/lib/rustdesk/rustdesk --tray
set 16 13:29:51 l200-com01.vr.local rustdesk[2874]: Sorry, user gdm is not allowed to execute '/usr/lib/rustdesk/rustdesk --tray' as gdm on l200-com01.vr.local.
set 16 13:29:51 l200-com01.vr.local rustdesk[2888]: Cannot load libcuda.so.1
set 16 13:29:51 l200-com01.vr.local rustdesk[2888]: Cannot load libcuda.so.1
set 16 13:30:22 l200-com01.vr.local sudo[4830]:     root : PWD=/ ; USER=l200comercial01 ; ENV=XDG_RUNTIME_DIR=/run/user/632801162 ; COMMAND=/usr/lib/rustdesk/rustdesk --server
set 16 13:30:22 l200-com01.vr.local sudo[4830]: pam_unix(sudo:session): session opened for user l200comercial01(uid=632801162) by (uid=0)
set 16 13:30:22 l200-com01.vr.local sudo[4836]: l200comercial01 : command not allowed ; PWD=/ ; USER=l200comercial01 ; ENV=XDG_RUNTIME_DIR=/run/user/632801162 ; COMMAND=/usr/lib/rustde>
set 16 13:30:22 l200-com01.vr.local rustdesk[4836]: Sorry, user l200comercial01 is not allowed to execute '/usr/lib/rustdesk/rustdesk --tray'** as l200comercial01 on l200-com01.vr.l>
set 16 13:30:22 l200-com01.vr.local rustdesk[4857]: Cannot load libcuda.so.1
```

**Comment by thiagodalsoto** [comment]
lol
I managed to get around it with the command;
ALL     ALL=(ALL) NOPASSWD: **SETENV**: /usr/lib/rustdesk/rustdesk

I'll leave it here so that others can take advantage of it in the future :)

**Reply by huyz** [reply]
Does this have anything to do with the `Cannot load libcuda.so.1` error? Cause that's what I'm seeing (not the `user gdm is not allowed` error)
